upi/openstack/021_network.yaml

```diff
@@ -4,15 +4,57 @@
   gather_facts: no
 
   tasks:
+  - name: 'list network'
+    command:
+      cmd: "openstack network list --name {{ os_aci_containers_network }} --tag {{ cluster_id_tag }}"
+    when: os_networking_type == "CiscoACI"
+    register: node_network
+
+  - name: 'get nested domain infra vlan of network'
+    command:
+      cmd: "neutron net-show {{ os_aci_containers_network }} -f value -c apic:nested_domain_infra_vlan"
+    when:
+      - os_networking_type == "CiscoACI"
+      - node_network.stdout != ""
+    register: infra_vlan
+
+  - name: 'get nested domain service vlan of network'
+    command:
+      cmd: "neutron net-show {{ os_aci_containers_network }} -f value -c apic:nested_domain_service_vlan"
+    when:
+      - os_networking_type == "CiscoACI"
+      - node_network.stdout != ""
+    register: service_vlan
+
   - name: 'Create ACI containers node network'
     command:
       cmd: "neutron net-create {{ os_aci_containers_network }} --apic:nested-domain-name openshift-domain --apic:nested-domain-type openshift --apic:nested_domain_infra_vlan {{ aci_cni['infra_vlan'] }} --apic:nested_domain_service_vlan {{ aci_cni['service_vlan'] }}"
-    when: os_networking_type == "CiscoACI"
+    when:
+      - os_networking_type == "CiscoACI"
+      - (node_network.stdout == "" ) or
+        (infra_vlan is defined and infra_vlan.stdout|int != aci_cni['infra_vlan']) or
+        (service_vlan is defined and service_vlan.stdout|int != aci_cni['service_vlan'])
```

Comment:
1. Wanted to make sure if node_network.stdout != "" and other condition i.e.
 (infra_vlan is defined and infra_vlan.stdout|int != aci_cni['infra_vlan']) or
        (service_vlan is defined and service_vlan.stdout|int != aci_cni['service_vlan']) passes, will the neutron net-create command will be able to create the network with same name and changed vlan configuration ? . Is that scenario tested.
2. Secondly, if the above condition passes, we don't want 2 networks with the same name, as the expectation is to have a single network with a unique name. This can be an issue where we update machineset where we use the UUID of the network and accidentally can fetch the UUID of another network. So, if that's not the case every time and we have a single network then that's good but if not, we need to first clean the current one and make the new one

Reply:
yes neutron-net create command create network with same name and changed vlan configurations. but it fails
while creating subnet, in this case playbook executions halts with error  before we update the machineset for UUID of the network.
In updated commit explicit error condition is added to avoid such scenarios i.e creation of  node network with same name but with different vlan values. Similar changes is made for other resources.